### ISSUE TITLE
DD to DMS conversion appears flawed

### ISSUE THREAD

**Issue by svoop** [zverok/geo_coord#5]
Take the DD coordinates `22.2` = `22°12'00"` or `33.3` = `33°18'00"`:

```ruby
g = Geo::Coord.new(22.2, 33.3)
g.strfcoord('%latd°%latm′%lats″%lath %lngd°%lngm′%lngs″%lngh')
=> "22°12′720″N 33°17′60″E"
g.lats
=> 720.0
g.lngs
=> 59.99999999998545
```

The float resolution seems to mess up the calculation.

**Comment by svoop** [comment]
FYI, here's a lightweight implementation of `Float#to_dms` and `String#to_dd` for a pet project of mine:
https://github.com/svoop/aixm/blob/master/lib/aixm/refinements.rb#L20

**Comment by zverok** [comment]
720 was just a bug (fixed by dff05cb).

As about float resolutions, I concerned about it myself while using the library, but not totally sure what's the reasonable solution. Maybe using `BigDecimal` for internal storage? Or maybe there should be at least some trick to catch `17'60"` cases.

**Comment by zverok** [comment]
Here's experimental `bigdecimal` branch: https://github.com/zverok/geo_coord/tree/bigdecimal. With it (and `lats` bug fixed), your example works perfectly as expected:
```ruby
g = Geo::Coord.new(22.2, 33.3)
# => #<Geo::Coord 22.200000,33.300000> 
g.lats
# => 0.0 
g.strfcoord('%latd°%latm′%lats″%lath %lngd°%lngm′%lngs″%lngh')
# => "22°12′0″N 33°18′0″E" 
```

**Comment by svoop** [comment]
`BigDecimal` should help, but IMO is a rather nuclear option in terms of performance, memory etc.

Implementations I found in other languages solve the `17'60"` float rounding issue by simply checking for 60 minutes or seconds after the calc:

https://github.com/svoop/aixm/blob/master/lib/aixm/refinements.rb#L20

In any case, you should maybe add roundtrip tests DDDMMSS.SS -> D.D -> DDDMMSS.SS to your specs and cycle through all possible 129'240'000 values. This takes too long to be part of the regular test suite, but only running from 0 to 2 degrees should be okay. And the full circle from 0 to 359 degrees once manually just to be sure. :smile:

**Comment by zverok** [comment]
Well, for what I know, `BigDecimal` is a cleaner solution for the problem. In fact, what we want is not just pretty formatting, but, say, this should work too:
```ruby
g = Geo::Coord.parse("22°12′0″N 33°18′0″E")
# => #<Geo::Coord 22.200000,33.300000> 
g.lngm == 18
# => false with Float, true with BigDecimal
g.lngs == 0
# => false with Float, true with BigDecimal
```

The downsides of BigDecimal that I currently see are:
1. Performance penalty, both on memory and CPU
2. Weird `inspect`: what is most naturally 22.2, looks like `#<BigDecimal:55665d9efd90,'0.222E2',18(36)>` (almost totally unreadable) in Ruby 2.3, and `0.222e2` (better, but still not immediately readable) in 2.4+.

**Comment by zverok** [comment]
Fixed in 0.1.0
